### DISCUSSION TITLE
Should the IDE show the full type of a Selectable generated via a transparent macro?

### DISCUSSION THREAD

**Discussion by hmf** [scalameta/metals-vscode#1470] [Q&A]
I am trying to create structural types "dynamically" via `Selectable` using a transparent macro.  Say I have this:

```Scala
package examples.select

// From https://users.scala-lang.org/t/how-to-refine-type-dynamically-in-scala-3-whitebox-macro/9220

import scala.quoted.*

def refined_1(using Quotes) =
  '{
    new scala.reflect.Selectable:
      val field: String = "value"
  }

transparent inline def structure_1 = ${refined_1}

// Get Any instead of structural type
def refined_2(using Quotes) : Expr[Any] =
  '{
    new scala.reflect.Selectable:
      val field: String = "value"
  }

transparent inline def structure_2 = ${refined_2}

// Get Any instead of structural type
def refined_3(using Quotes) : Expr[Any] =
  import quotes.reflect.*
  val cl = '{
    new scala.reflect.Selectable:
      val field: String = "value"
  }
  val tree = cl.asTerm
  val typ = Refined.copy(tree)(
    TypeTree.of[Object],
    List(ValDef.copy(tree)("field", TypeIdent.copy(tree)("String"), None))
  )
  // We get Any
  // Typed(tree, typ).asExpr
  val typedTerm = Typed(tree, typ)
  typedTerm.tpe.asType match
    case '[t] => typedTerm.asExprOf[t]

transparent inline def structure_3 = ${refined_3}

```
Now I use these so:

```Scala
package examples.select


val x: String = structure_1.field
val y: String = structure_2.field
val z: String = structure_3.field

val f1 = structure_1.field
val f2 = structure_3.field
```

It compiles ok. The IDE allows me to peek and got to the correct types/implementations. However when I hover over the refined type and the fields or assigned variables, I get not pop-up with the information.

So my question is, is this the expected behavior or should we see the types. I was hopping the latter was possible. I also would like to now if code suggestions should work in the first case. So of I write `structure_1.`, the IDE should show `field`. 

TIA

**Comment by hmf** [comment]
Very strange. I exited and reopened the project and now type information appears correctly.  I was able to capture the example below. I have messed around with the code and it still works. I am also able to use intellisense with no problems. 

Apologies for the noise, but I don't understand why it was not working before. If I figure this out I will report back.

Thanks

![Screenshot from 2024-03-08 16-02-31](https://github.com/scalameta/metals-vscode/assets/1138178/b80c8b2c-5e65-4845-ae1f-10f0aa9956fd)

**Reply by tgodzik** [reply]
There could be some reports in `.metals/reports` indicating why the problems before were happening. Would be great to figure out what was wrong.

**Reply by hmf** [reply]
I found a a few pertaining to the code I described above. 

I seem to have about 24 of these files all with the same time stamp in `.metals/.reports/metals-full` and what looks like the same content. Here is the report. 

[r_compiler-error_(meta)_15-23-26-277.md](https://github.com/scalameta/metals-vscode/files/14541320/r_compiler-error_.meta._15-23-26-277.md)
